### ISSUE TITLE
Wrong behaviour...!

### ISSUE THREAD

**Issue by isaac-ramthal-css** [BobKnothe/autoNumeric#191]
Hi,

I have the following code and when i enter a value _15 000 000 000.45_ and put the cursor on first zero and press backspace, the plugin behaves strange and removes all of the later numbers reducing the number to _150.45_. I am using 1.9.43 and Firefox 44.0

```
<!DOCTYPE HTML>
<html>
<head>
    <meta charset="utf-8">
    <title>autoNumeric</title>
    <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.10.1/jquery.min.js"></script>
    <script type="text/javascript" src="autoNumeric-min.js"></script>
    <script type="text/javascript">
        $(document).on('focus', '.currency', function(e){
            if ($(this).hasClass('autoNumeric') === false) {
                console.log('autoNumeric binded');
                $(this).addClass('autoNumeric').autoNumeric('init', {aSep: ' '});
            }
        });
    </script>
</head>

<body>
<div id="main" style="width:900px; border:#000000; background-color:#FFFFFF">
    <div id="content" style="padding: 5px 10px 10px 10px; margin-top:0px">
        <form name="form1" method="post" action="">
            <table width="880" border="0">
                <tr>
                    <td width="33%"><input class="currency" type="text" class="auto" style="text-align:right;"></td>
                </tr>
            </table>
        </form>
    </div>
</div>
</body>
</html>
`

**Comment by BobKnothe** [comment]
Issac,

What you are encountering is the limits of javaScript numeric accuracy which is well documented.

I suggest trying my version 2.0 (even though is say beta it is stable) which has support for Big.js (https://github.com/MikeMcl/big.js/) that should address your issue and still maintain precision min/max values.

Please let me know the results.

Bob

**Comment by isaac-ramthal-css** [comment]
@BobKnothe Thanks for the reply but oddly enough I got the wrong behaviour only when using the minified version of the library.  I switched to the uncompressed version and it ran perfectly.  I am not sure if there was some issue in the minified version which I was using (I fetched it using bower). 

I thought it was the way I am binding the plugin that is creating the issue.  I will try to recreate the issue on fiddle later and try posting an example.  But it is working great for now.  Thanks for creating an awesome plugin.

**Comment by BobKnothe** [comment]
Isaac,
 
Thanks for the info - look forward to seeing the fiddle.
 
I may need to try a different compressor.
 
Bob
 
--------- Original Message --------- Subject: Re: [autoNumeric] Wrong behaviour...! (#191)

**Comment by isaac-ramthal-css** [comment]
@BobKnothe Sorry strangely could not reproduce this on jsfiddle so attaching a miniproject. Try this in Firefox.  As of this date it does not work even in latest version of firefox 44.0.2.

[autoNumeric.zip](https://github.com/BobKnothe/autoNumeric/files/133769/autoNumeric.zip)

**Comment by BobKnothe** [comment]
Isaac,
 
Looks like there is two issues going on:
 
1) This is my error - it looks I minimized version 2.0 beta and labeled it 1.9.43 
2) This one is a little more complicated. On "focusin" you initialize autoNumeric which on 1.9.43 is OK, but on version 2.0 a internal setting is modified on the "focusin". The problem is you are initializing autoNumeric after the "focusin" event so the internal setting is not properly changed.
 
So first I can re-minimize version 1.9.43 and properly label.
On version 2.0 I can add a line on the "keydown" event on that modifies the internal settings. My initial tests shows that it works but it needs more testing before releasing.
 
I hope you understand my explanation.
 
Bob  
 
--------- Original Message --------- Subject: Re: [autoNumeric] Wrong behaviour...! (#191)

**Comment by isaac-ramthal-css** [comment]
@BobKnothe  Got it :)  Hoping to see new patch for this soon.  I am using 1.9.x.
